### PR TITLE
Add support for Cloudwatch Event API Destination and Connections

### LINKED ISSUE #17950: Amazon EventBridge introduces support for API Destinations
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

> Amazon EventBridge now supports API Destinations, a simple and reliable way for customers to send events to any HTTP API

### New or Affected Resource(s)

* `aws_cloudwatch_event_api_destination`
* `aws_cloudwatch_event_connection`

### Potential Terraform Configuration

```hcl
resource "aws_cloudwatch_event_api_destination" ""example" {
  name                = "example-api-destination"
  connection_arn      = aws_cloudwatch_event_connection.example.arn
  invocation_endpoint = "http://example.com/"
  http_method         = "GET"
}

resource "aws_cloudwatch_event_connection" ""example" {
  name = "example-connection"

 # ...
}
```

### References

[Announcement](https://aws.amazon.com/about-aws/whats-new/2021/03/amazon-eventbridge-introduces-support-for-api-destinations/).
[Blog post](https://aws.amazon.com/blogs/compute/using-api-destinations-with-amazon-eventbridge/).

Requires [AWS SDK v1.37.24](https://github.com/aws/aws-sdk-go/releases/tag/v1.37.24):

* #17934

### PR DESCRIPTION
**Pull request by jordanfinners**
Add support for Cloudwatch Event API Destination and Connections (aka Eventbridge API Destination and Connections)

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #17950

Output from acceptance testing:

Connection:
```
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSCloudWatchEventConnection -timeout 180m
=== RUN   TestAccAWSCloudWatchEventConnection_apiKey
=== PAUSE TestAccAWSCloudWatchEventConnection_apiKey
=== RUN   TestAccAWSCloudWatchEventConnection_basic
=== PAUSE TestAccAWSCloudWatchEventConnection_basic
=== RUN   TestAccAWSCloudWatchEventConnection_oAuth
=== PAUSE TestAccAWSCloudWatchEventConnection_oAuth
=== RUN   TestAccAWSCloudWatchEventConnection_invocationHttpParameters
=== PAUSE TestAccAWSCloudWatchEventConnection_invocationHttpParameters
=== RUN   TestAccAWSCloudWatchEventConnection_disappears
=== PAUSE TestAccAWSCloudWatchEventConnection_disappears
=== CONT  TestAccAWSCloudWatchEventConnection_apiKey
=== CONT  TestAccAWSCloudWatchEventConnection_invocationHttpParameters
=== CONT  TestAccAWSCloudWatchEventConnection_disappears
=== CONT  TestAccAWSCloudWatchEventConnection_oAuth
=== CONT  TestAccAWSCloudWatchEventConnection_basic
--- PASS: TestAccAWSCloudWatchEventConnection_disappears (52.12s)
--- PASS: TestAccAWSCloudWatchEventConnection_invocationHttpParameters (83.60s)
--- PASS: TestAccAWSCloudWatchEventConnection_basic (86.43s)
--- PASS: TestAccAWSCloudWatchEventConnection_apiKey (90.90s)
--- PASS: TestAccAWSCloudWatchEventConnection_oAuth (92.36s)
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws       92.449s
```

API Destination:
```
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSCloudWatchEventApiDestination -timeout 180m
=== RUN   TestAccAWSCloudWatchEventApiDestination_basic
=== PAUSE TestAccAWSCloudWatchEventApiDestination_basic
=== RUN   TestAccAWSCloudWatchEventApiDestination_optional
=== PAUSE TestAccAWSCloudWatchEventApiDestination_optional
=== RUN   TestAccAWSCloudWatchEventApiDestination_disappears
=== PAUSE TestAccAWSCloudWatchEventApiDestination_disappears
=== CONT  TestAccAWSCloudWatchEventApiDestination_basic
=== CONT  TestAccAWSCloudWatchEventApiDestination_optional
=== CONT  TestAccAWSCloudWatchEventApiDestination_disappears
--- PASS: TestAccAWSCloudWatchEventApiDestination_disappears (34.21s)
--- PASS: TestAccAWSCloudWatchEventApiDestination_basic (76.61s)
--- PASS: TestAccAWSCloudWatchEventApiDestination_optional (83.28s)
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws       83.359s
```

Tests for the additional data source
```
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSDataSourceCloudwatch_Event_Connection_basic -timeout 180m
=== RUN   TestAccAWSDataSourceCloudwatch_Event_Connection_basic
=== PAUSE TestAccAWSDataSourceCloudwatch_Event_Connection_basic
=== CONT  TestAccAWSDataSourceCloudwatch_Event_Connection_basic
--- PASS: TestAccAWSDataSourceCloudwatch_Event_Connection_basic (23.06s)
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws       23.111s
```

### CODE REVIEW

**Comment by jordanfinners** [comment]
Currently working on the tests working, would someone be able to check over the schema structure for the Connection as its pretty nested 🙏

**Comment by YakDriver** [comment]
@jordanfinners I skimmed over the schema and didn't see anything glaringly wrong. It is quite nested but you seem to be managing it well. If at all possible, we do like the arguments to be in alphabetical order. With so many people adding so many things, keeping them in order helps.

I would also highly recommend looking at https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/data-handling-and-conversion.md for information on current best practices. Unfortunately, we have a lot of places in the codebase with deprecated styles and ways of doing things. If you're following those places, it does make review harder. Another good one that's up to date is here: https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/retries-and-waiters.md. 

Best regards, and best of luck!

**Comment by jordanfinners** [comment]
> Best regards, and best of luck!

Thanks @YakDriver !
I kept the parameters in the same order as the AWS SDK so it matches it if anyone else looks at it.

One issue I'm having with the tests is that I can't get some of the values out when describing the resource e.g. ClientSecret, Password etc.
So when I'm running the tests I get the following:
```
resource_aws_cloudwatch_event_connection_test.go:81: Step 1/4 error: After applying this test step, the plan was not empty.
        stdout:
        
        
        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place
        
        Terraform will perform the following actions:
        
          # aws_cloudwatch_event_connection.api_key will be updated in-place
          ~ resource "aws_cloudwatch_event_connection" "api_key" {
                id                 = "tf-acc-test-8524034810703668716"
                name               = "tf-acc-test-8524034810703668716"
                # (3 unchanged attributes hidden)
        
              ~ auth_parameters {
                  ~ api_key {
                      + value = (sensitive value)
                        # (1 unchanged attribute hidden)
                    }
                }
            }
        
        Plan: 0 to add, 1 to change, 0 to destroy.
```

Do you have any advice on how to deal with this?

**Comment by jordanfinners** [comment]
Hey @YakDriver, quick question the ARN for both API Destinations and Connections has a random UUID at the end e.g. `arn:aws:events:us-west-2:accountID:api-destination/nameOfResource/eb8ea254-2b82-4a62-9d7c-0c70036e182e`

And I'm not sure how to handle this in the following test(s) `testAccGetPartitiontestAccCheckResourceAttrRegionalARN(resourceName, "arn", "events", fmt.Sprintf("event-ApiDestination/%s", name))` 

I could create a separate method to regex it but wondered if you had any better suggestions? 🙏

**Comment by jordanfinners** [comment]
> @jordanfinners I skimmed over the schema and didn't see anything glaringly wrong. It is quite nested but you seem to be managing it well. If at all possible, we do like the arguments to be in alphabetical order. With so many people adding so many things, keeping them in order helps.
> 
> I would also highly recommend looking at https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/data-handling-and-conversion.md for information on current best practices. Unfortunately, we have a lot of places in the codebase with deprecated styles and ways of doing things. If you're following those places, it does make review harder. Another good one that's up to date is here: https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/retries-and-waiters.md.
> 
> Best regards, and best of luck!

@YakDriver ready for review! 👍

File: website/docs/r/cloudwatch_event_connection.html.markdown [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `arn` - The Amazon Resource Name (ARN) of the event bus.
+
+
+## Import
+
+EventBridge event buses can be imported using the `name`, e.g.
```
Comment (julesjcraske) [inline website/docs/r/cloudwatch_event_connection.html.markdown:196]: This is a copy paste error from the event bus page.

Comment (jordanfinners) [reply website/docs/r/cloudwatch_event_connection.html.markdown]: Great spot! Fixed :)

**Comment by jordanfinners** [comment]
Hey @YakDriver would you mind reviewing this again please 🙏  Would be great to get it merged 😃

**Comment by jordanfinners** [comment]
Hey @heitorlessa did you have chance to review this please 🙏

**Comment by heitorlessa** [comment]
Destination resource LGTM. I need some more time to digest the Connection resource where I can spend 30 min uninterruptedly to double check some Hashicorp common practices on flattening

File: aws/resource_aws_cloudwatch_event_connection.go [hunk 2]
```diff
@@ -0,0 +1,785 @@
...
+						"invocation_http_parameters": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem:     connectionHttpParameters,
+						},
+					},
+				},
+			},
+			"arn": {
```
Comment (heitorlessa) [inline aws/resource_aws_cloudwatch_event_connection.go:224]: could you also add `secret_arn` as that will be returned if any secret value is passed for params?

File: aws/resource_aws_cloudwatch_event_connection.go [hunk 3]
```diff
@@ -0,0 +1,785 @@
...
+		d.SetId("")
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("error reading CloudWatchEvent connection: %w", err)
+	}
+
+	log.Printf("[DEBUG] Found CloudWatchEvent connection: %#v", *output)
+
+	d.Set("arn", output.ConnectionArn)
```
Comment (heitorlessa) [inline aws/resource_aws_cloudwatch_event_connection.go:284]: Please also include `secret_arn`: https://docs.aws.amazon.com/sdk-for-go/api/service/cloudwatchevents/#CreateConnectionOutput

File: website/docs/r/cloudwatch_event_connection.html.markdown [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+* `query_string` - (Optional) Contains additional query string parameters for the connection. You can include up to 100 additional body string parameters per request. Each additional parameter counts towards the event payload size, which cannot exceed 64 KB. Each parameter can contain the following:
+    * `key` - (Required) The key for the parameter.
+    * `value` - (Required) The value associated with the key. Created and stored in AWS Secrets Manager if is secret.
+    * `is_value_secret` - (Optional) Specified whether the value is secret.
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `arn` - The Amazon Resource Name (ARN) of the event bus.
```
Comment (heitorlessa) [inline website/docs/r/cloudwatch_event_connection.html.markdown:191]: could you please add `secret_arn`?

**Review by heitorlessa — CHANGES_REQUESTED** [review]
Looks pretty good except two things:

1. **`secret_arn` attribute not being set**. When parameter values are set to be secret EventBridge return a Secret ARN that it stored in Secrets Manager so one could fetch it. 

2. **Data source** for `event_connection` since that can be used by one or more API Destinations. This will help customers using TF Modules or retrieving an existing connection instead of hardcoding its ARN.

Once these are done, I'll pass this in for priorisation and merge :) 

Thanks a lot for doing this, it looks great!

File: aws/resource_aws_cloudwatch_event_connection.go (hunk 2 above)
Comment (jordanfinners) [reply aws/resource_aws_cloudwatch_event_connection.go:224]: Will you only get one `secret_arn` back or will it be one per param?

**Comment by heitorlessa** [comment]
One secret_arn per connection for an arbitrary number of params with a
secret value.

**Comment by jordanfinners** [comment]
> Looks pretty good except two things:
> 
> 1. **`secret_arn` attribute not being set**. When parameter values are set to be secret EventBridge return a Secret ARN that it stored in Secrets Manager so one could fetch it.
> 2. **Data source** for `event_connection` since that can be used by one or more API Destinations. This will help customers using TF Modules or retrieving an existing connection instead of hardcoding its ARN.
> 
> Once these are done, I'll pass this in for priorisation and merge :)
> 
> Thanks a lot for doing this, it looks great!

I've added these now! :D

**Review by heitorlessa — APPROVED** [review]
Thanks for making these changes - LGTM ;)

**Comment by jordanfinners** [comment]
Not sure what github did, was re-requesting review from julesjcraske and it removed `request for review from hashicorp/terraform-aws` :/

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

##### Commercial

```console
% make testacc TEST=./aws TESTARGS='-run=TestAccAWSCloudWatchEventConnection_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSCloudWatchEventConnection_ -timeout 180m
=== RUN   TestAccAWSCloudWatchEventConnection_apiKey
=== PAUSE TestAccAWSCloudWatchEventConnection_apiKey
=== RUN   TestAccAWSCloudWatchEventConnection_basic
=== PAUSE TestAccAWSCloudWatchEventConnection_basic
=== RUN   TestAccAWSCloudWatchEventConnection_oAuth
=== PAUSE TestAccAWSCloudWatchEventConnection_oAuth
=== RUN   TestAccAWSCloudWatchEventConnection_invocationHttpParameters
=== PAUSE TestAccAWSCloudWatchEventConnection_invocationHttpParameters
=== RUN   TestAccAWSCloudWatchEventConnection_disappears
=== PAUSE TestAccAWSCloudWatchEventConnection_disappears
=== CONT  TestAccAWSCloudWatchEventConnection_apiKey
=== CONT  TestAccAWSCloudWatchEventConnection_invocationHttpParameters
=== CONT  TestAccAWSCloudWatchEventConnection_disappears
=== CONT  TestAccAWSCloudWatchEventConnection_oAuth
=== CONT  TestAccAWSCloudWatchEventConnection_basic
--- PASS: TestAccAWSCloudWatchEventConnection_disappears (34.57s)
--- PASS: TestAccAWSCloudWatchEventConnection_basic (116.08s)
--- PASS: TestAccAWSCloudWatchEventConnection_invocationHttpParameters (116.51s)
--- PASS: TestAccAWSCloudWatchEventConnection_oAuth (125.83s)
--- PASS: TestAccAWSCloudWatchEventConnection_apiKey (167.92s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	171.076s
% make testacc TEST=./aws TESTARGS='-run=TestAccAWSCloudWatchEventApiDestination_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSCloudWatchEventApiDestination_ -timeout 180m
=== RUN   TestAccAWSCloudWatchEventApiDestination_basic
=== PAUSE TestAccAWSCloudWatchEventApiDestination_basic
=== RUN   TestAccAWSCloudWatchEventApiDestination_optional
=== PAUSE TestAccAWSCloudWatchEventApiDestination_optional
=== RUN   TestAccAWSCloudWatchEventApiDestination_disappears
=== PAUSE TestAccAWSCloudWatchEventApiDestination_disappears
=== CONT  TestAccAWSCloudWatchEventApiDestination_basic
=== CONT  TestAccAWSCloudWatchEventApiDestination_disappears
=== CONT  TestAccAWSCloudWatchEventApiDestination_optional
--- PASS: TestAccAWSCloudWatchEventApiDestination_disappears (14.43s)
--- PASS: TestAccAWSCloudWatchEventApiDestination_optional (36.85s)
--- PASS: TestAccAWSCloudWatchEventApiDestination_basic (37.66s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	40.878s
% make testacc TEST=./aws TESTARGS='-run=TestAccAWSDataSourceCloudwatch_Event_Connection_' 
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSDataSourceCloudwatch_Event_Connection_ -timeout 180m
=== RUN   TestAccAWSDataSourceCloudwatch_Event_Connection_basic
=== PAUSE TestAccAWSDataSourceCloudwatch_Event_Connection_basic
=== CONT  TestAccAWSDataSourceCloudwatch_Event_Connection_basic
--- PASS: TestAccAWSDataSourceCloudwatch_Event_Connection_basic (75.98s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	79.013s
```

##### GovCloud

```console
% make testacc TEST=./aws TESTARGS='-run=TestAccAWSCloudWatchEventConnection_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSCloudWatchEventConnection_ -timeout 180m
=== RUN   TestAccAWSCloudWatchEventConnection_apiKey
=== PAUSE TestAccAWSCloudWatchEventConnection_apiKey
=== RUN   TestAccAWSCloudWatchEventConnection_basic
=== PAUSE TestAccAWSCloudWatchEventConnection_basic
=== RUN   TestAccAWSCloudWatchEventConnection_oAuth
=== PAUSE TestAccAWSCloudWatchEventConnection_oAuth
=== RUN   TestAccAWSCloudWatchEventConnection_invocationHttpParameters
=== PAUSE TestAccAWSCloudWatchEventConnection_invocationHttpParameters
=== RUN   TestAccAWSCloudWatchEventConnection_disappears
=== PAUSE TestAccAWSCloudWatchEventConnection_disappears
=== CONT  TestAccAWSCloudWatchEventConnection_apiKey
=== CONT  TestAccAWSCloudWatchEventConnection_invocationHttpParameters
=== CONT  TestAccAWSCloudWatchEventConnection_oAuth
=== CONT  TestAccAWSCloudWatchEventConnection_disappears
=== CONT  TestAccAWSCloudWatchEventConnection_basic
=== CONT  TestAccAWSCloudWatchEventConnection_disappears
    provider_test.go:1103: skipping test for aws-us-gov/us-gov-west-1: Error running apply: exit status 1
        2021/05/24 13:48:42 [DEBUG] Using modified User-Agent: Terraform/0.12.31 HashiCorp-terraform-exec/0.13.3
        
        Error: Creating CloudWatchEvent connection (tf-acc-test-4305597680382262485) failed: UnknownOperationException: Operation is disabled in this region.
        	status code: 400, request id: d0907a45-332c-49ac-b3fe-fbc755b4795f
        
          on terraform_plugin_test.tf line 2, in resource "aws_cloudwatch_event_connection" "api_key":
           2: resource "aws_cloudwatch_event_connection" "api_key" {
        
        
=== CONT  TestAccAWSCloudWatchEventConnection_basic
    provider_test.go:1103: skipping test for aws-us-gov/us-gov-west-1: Error running apply: exit status 1
        2021/05/24 13:48:42 [DEBUG] Using modified User-Agent: Terraform/0.12.31 HashiCorp-terraform-exec/0.13.3
        
        Error: Creating CloudWatchEvent connection (tf-acc-test-9201948011675794539) failed: UnknownOperationException: Operation is disabled in this region.
        	status code: 400, request id: f0cab8e7-efef-4da5-9d3e-0af624277624
        
          on terraform_plugin_test.tf line 2, in resource "aws_cloudwatch_event_connection" "basic":
           2: resource "aws_cloudwatch_event_connection" "basic" {
        
        
=== CONT  TestAccAWSCloudWatchEventConnection_oAuth
    provider_test.go:1103: skipping test for aws-us-gov/us-gov-west-1: Error running apply: exit status 1
        2021/05/24 13:48:42 [DEBUG] Using modified User-Agent: Terraform/0.12.31 HashiCorp-terraform-exec/0.13.3
        
        Error: Creating CloudWatchEvent connection (tf-acc-test-5998789718151218737) failed: UnknownOperationException: Operation is disabled in this region.
        	status code: 400, request id: 60897055-390a-4c23-a8ef-da18e0a5fa3f
        
          on terraform_plugin_test.tf line 2, in resource "aws_cloudwatch_event_connection" "oauth":
           2: resource "aws_cloudwatch_event_connection" "oauth" {
        
        
=== CONT  TestAccAWSCloudWatchEventConnection_invocationHttpParameters
    provider_test.go:1103: skipping test for aws-us-gov/us-gov-west-1: Error running apply: exit status 1
        2021/05/24 13:48:42 [DEBUG] Using modified User-Agent: Terraform/0.12.31 HashiCorp-terraform-exec/0.13.3
        
        Error: Creating CloudWatchEvent connection (tf-acc-test-5357976610980534660) failed: UnknownOperationException: Operation is disabled in this region.
        	status code: 400, request id: 5b69a983-0b51-4846-b044-298a6d7ccc2f
        
          on terraform_plugin_test.tf line 2, in resource "aws_cloudwatch_event_connection" "invocation_http_parameters":
           2: resource "aws_cloudwatch_event_connection" "invocation_http_parameters" {
        
        
=== CONT  TestAccAWSCloudWatchEventConnection_apiKey
    provider_test.go:1103: skipping test for aws-us-gov/us-gov-west-1: Error running apply: exit status 1
        2021/05/24 13:48:42 [DEBUG] Using modified User-Agent: Terraform/0.12.31 HashiCorp-terraform-exec/0.13.3
        
        Error: Creating CloudWatchEvent connection (tf-acc-test-1206077227320570848) failed: UnknownOperationException: Operation is disabled in this region.
        	status code: 400, request id: 7c7b1d04-a636-4383-83a4-8c5184bbd379
        
          on terraform_plugin_test.tf line 2, in resource "aws_cloudwatch_event_connection" "api_key":
           2: resource "aws_cloudwatch_event_connection" "api_key" {
        
        
--- SKIP: TestAccAWSCloudWatchEventConnection_disappears (6.77s)
--- SKIP: TestAccAWSCloudWatchEventConnection_basic (6.77s)
--- SKIP: TestAccAWSCloudWatchEventConnection_oAuth (6.79s)
--- SKIP: TestAccAWSCloudWatchEventConnection_apiKey (6.82s)
--- SKIP: TestAccAWSCloudWatchEventConnection_invocationHttpParameters (6.82s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	9.961s
% make testacc TEST=./aws TESTARGS='-run=TestAccAWSCloudWatchEventApiDestination_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSCloudWatchEventApiDestination_ -timeout 180m
=== RUN   TestAccAWSCloudWatchEventApiDestination_basic
=== PAUSE TestAccAWSCloudWatchEventApiDestination_basic
=== RUN   TestAccAWSCloudWatchEventApiDestination_optional
=== PAUSE TestAccAWSCloudWatchEventApiDestination_optional
=== RUN   TestAccAWSCloudWatchEventApiDestination_disappears
=== PAUSE TestAccAWSCloudWatchEventApiDestination_disappears
=== CONT  TestAccAWSCloudWatchEventApiDestination_basic
=== CONT  TestAccAWSCloudWatchEventApiDestination_disappears
=== CONT  TestAccAWSCloudWatchEventApiDestination_optional
=== CONT  TestAccAWSCloudWatchEventApiDestination_disappears
    provider_test.go:1103: skipping test for aws-us-gov/us-gov-west-1: Error running apply: exit status 1
        2021/05/24 13:54:49 [DEBUG] Using modified User-Agent: Terraform/0.12.31 HashiCorp-terraform-exec/0.13.3
        
        Error: Creating CloudWatchEvent connection (tf-acc-test-5609326720649779595) failed: UnknownOperationException: Operation is disabled in this region.
        	status code: 400, request id: efdfbc38-8c65-4a45-ad08-328027f3c09f
        
          on terraform_plugin_test.tf line 9, in resource "aws_cloudwatch_event_connection" "test":
           9: resource "aws_cloudwatch_event_connection" "test" {
        
        
=== CONT  TestAccAWSCloudWatchEventApiDestination_basic
    provider_test.go:1103: skipping test for aws-us-gov/us-gov-west-1: Error running apply: exit status 1
        2021/05/24 13:54:49 [DEBUG] Using modified User-Agent: Terraform/0.12.31 HashiCorp-terraform-exec/0.13.3
        
        Error: Creating CloudWatchEvent connection (tf-acc-test-2705877033536899691) failed: UnknownOperationException: Operation is disabled in this region.
        	status code: 400, request id: d1efbc3d-1aac-4b33-b63f-4c5bebd651e2
        
          on terraform_plugin_test.tf line 9, in resource "aws_cloudwatch_event_connection" "test":
           9: resource "aws_cloudwatch_event_connection" "test" {
        
        
=== CONT  TestAccAWSCloudWatchEventApiDestination_optional
    provider_test.go:1103: skipping test for aws-us-gov/us-gov-west-1: Error running apply: exit status 1
        2021/05/24 13:54:49 [DEBUG] Using modified User-Agent: Terraform/0.12.31 HashiCorp-terraform-exec/0.13.3
        
        Error: Creating CloudWatchEvent connection (tf-acc-test-4958863464540868710) failed: UnknownOperationException: Operation is disabled in this region.
        	status code: 400, request id: 3f14ebad-dce3-46be-b1a0-07e37eb8ec04
        
          on terraform_plugin_test.tf line 12, in resource "aws_cloudwatch_event_connection" "test":
          12: resource "aws_cloudwatch_event_connection" "test" {
        
        
--- SKIP: TestAccAWSCloudWatchEventApiDestination_disappears (5.88s)
--- SKIP: TestAccAWSCloudWatchEventApiDestination_basic (5.88s)
--- SKIP: TestAccAWSCloudWatchEventApiDestination_optional (5.89s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	9.461s
% make testacc TEST=./aws TESTARGS='-run=TestAccAWSDataSourceCloudwatch_Event_Connection_' 
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSDataSourceCloudwatch_Event_Connection_ -timeout 180m
=== RUN   TestAccAWSDataSourceCloudwatch_Event_Connection_basic
=== PAUSE TestAccAWSDataSourceCloudwatch_Event_Connection_basic
=== CONT  TestAccAWSDataSourceCloudwatch_Event_Connection_basic
    provider_test.go:1146: skipping test for aws-us-gov/us-gov-west-1: Error running apply: exit status 1
        2021/05/24 13:59:08 [DEBUG] Using modified User-Agent: Terraform/0.12.31 HashiCorp-terraform-exec/0.13.3
        
        Error: Creating CloudWatchEvent connection (tf-acc-test-3577340883533182219) failed: UnknownOperationException: Operation is disabled in this region.
        	status code: 400, request id: f1668189-877c-400e-a2e5-c140fd6c5b52
        
          on terraform_plugin_test.tf line 2, in resource "aws_cloudwatch_event_connection" "api_key":
           2: resource "aws_cloudwatch_event_connection" "api_key" {
        
        
--- SKIP: TestAccAWSDataSourceCloudwatch_Event_Connection_basic (6.56s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	9.614s
```

**Comment by ewbankkit** [comment]
@jordanfinners Thanks for the contribution 👏 🎉.
@heitorlessa @DrFaust92 @julesjcraske Thanks for the reviews.

Whilst running the acceptance tests I hit:

```
    resource_aws_cloudwatch_event_connection_test.go:283: Step 3/4 error: Error running apply: exit status 1
        2021/05/24 13:29:43 [DEBUG] Using modified User-Agent: Terraform/0.12.31 HashiCorp-terraform-exec/0.13.3
        
        Error: Error deleting CloudWatchEvent connection (tf-acc-test-8781869599888619638): ConcurrentModificationException: Failed to delete the connection(s). Connection 'tf-acc-test-8781869599888619638' was modified concurrently.
```

in `TestAccAWSCloudWatchEventConnection_oAuth` so I went ahead and added waiters in the resource creation and update handlers.

**Pull request merged**